pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -583,13 +582,10 @@ func (qjm *XController) getAppWrapperCompletionStatus(caw *arbv1.AppWrapper) arb
 	for i, genericItem := range caw.Spec.AggrResources.GenericItems {
 		if len(genericItem.CompletionStatus) > 0 {
 			objectName := genericItem.GenericTemplate
-			var unstruct unstructured.Unstructured
-			unstruct.Object = make(map[string]interface{})
-			var blob interface{}
-			if err := jsons.Unmarshal(objectName.Raw, &blob); err != nil {
-				klog.Errorf("[getAppWrapperCompletionStatus] Error unmarshalling, err=%#v", err)
+			unstruct, err := genericresource.UnmarshalToUnstructured(objectName.Raw)
+			if err != nil {
+			    klog.Errorf("[getAppWrapperCompletionStatus] Error: %v", err)
```

Comment:
Does it make sense to continue the loop iteration if `err != nil`?

Reply:
Thanks for the feedback @ronensc :)

You are correct, we could exit the loop here, as if there is an error returned the remainder of the logic wont be operational. 

However, I would be hesitant making adjustments in this pr, It may be worth looking into error handling overall separately.